tornado/test/simple_httpclient_test.py

```diff
@@ -322,12 +330,11 @@ def test_options_request(self):
     def test_no_content(self):
         response = self.fetch("/no_content")
         self.assertEqual(response.code, 204)
-        # 204 status doesn't need a content-length, but tornado will
-        # add a zero content-length anyway.
+        # 204 status shouldn't have a content-length
         #
         # A test without a content-length header is included below
         # in HTTP204NoContentTestCase.
```

Comment:
There are three cases: no header (fine), `Content-Length: 0` (fine), and non-zero content-length (error). We no longer have a case demonstrating the acceptance of `Content-Length: 0`, and the error case is made awkward by the need to write directly to the connection from a RequestHandler (which I think will break things in http2). I'm suggesting that this test just cover the simple case (no header), and both cases in which the header is present move to HTTP204NoContentTestCase.